Technico/Services/PropertyService.cs

```diff
@@ -0,0 +1,112 @@
+﻿using Technico.Models;
+using Technico.Repositories;
+using Technico.Dtos;
+
+namespace Technico.Services;
+
+public class PropertyService 
+{
+    private readonly PropertyRepository _propertyRepository;
+
+    public PropertyService(PropertyRepository propertyRepository)
+    {
+        _propertyRepository = propertyRepository;
+    }
+
+    public async Task<PropertyDTO?> CreateAsync(PropertyDTO propertyDTO)
+    {
+        Console.WriteLine($"Input PropertyDTO - Address: {propertyDTO.Address}, OwnerID: {propertyDTO.OwnerID}");
+
+        var properties = await _propertyRepository.GetAllAsync();
+        bool PropertyExists = properties.Any(u => u.Address == propertyDTO.Address);
```

Comment:
Instead of retrieving all properties in CreateAsync and UpdateAsync, query the database directly to check for duplicates. For example:
public async Task<bool> AnyAsync(Expression<Func<Property, bool>> predicate)
{
    return await _dbContext.Properties.AnyAsync(predicate);
}